test/controllers/api/accounts_controller_test.rb

```diff
@@ -273,6 +273,74 @@ class Api::AccountsControllerTest < ActionController::TestCase
       assert_equal 'test', account['cleartext_username']
       assert_equal 'password', account['cleartext_password']
     end
+
+  end
+
+  context '#update' do
+    test 'update account with valid params structure' do
+      request.headers['Authorization-User'] = bob.username
+      request.headers['Authorization-Password'] = Base64.encode64('password')
+
+      account = accounts(:account1)
+
+      request_account = {
+        id: account.id,
+        account: {
+          accountname: "Bob Meyer",
+          tag: "taggy"
+        }
+      }
+      patch :update, params: request_account , xhr: true
+
+      assert_equal "Bob Meyer", Account.find(account.id).accountname
+      assert_equal "taggy", Account.find(account.id).tag
+      assert_equal 200, response.status
+    end
+
+    test 'update account without valid params structure should not occur' do
+      request.headers['Authorization-User'] = bob.username
+      request.headers['Authorization-Password'] = Base64.encode64('password')
+
+      account = accounts(:account1)
+
+      account.save()
+
+      request_account = {
+        id: account.id,
+        account: {
+          not_valid_key: "Bob Meyer",
```

Comment:
use password, username attrs here !
you wanna make sure no one can update them directly !

Reply:
use password, username as attr for this test !
make sure nobody can override this attrs directly in the db !

Reply:
title for test: "account's password and username attributes cannot be set by params"